easybuild/easyblocks/l/llvm.py

```diff
@@ -792,7 +790,17 @@ def configure_step(self):
         ]
         apply_regex_substitutions(lit_cfg_file, regex_subs)
 
-        self._set_gcc_prefix()
+        # For LLVM 18+ config files should be used and this option is deprecated and causes an error in 19
+        # But the --gcc-toolchain and --gcc-install-dir for flang are not supported before LLVM 19
+        # https://github.com/llvm/llvm-project/pull/87360
+        if LooseVersion(self.version) < '19':
+            self.log.debug("Using GCC_INSTALL_PREFIX")
+            self.general_opts['GCC_INSTALL_PREFIX'] = self.gcc_root
+        else:
+            # See https://github.com/llvm/llvm-project/pull/85891#issuecomment-2021370667
+            self.log.debug("Using '--gcc-install-dir' in CMAKE_C_FLAGS and CMAKE_CXX_FLAGS")
+            self.runtimes_cmake_args['CMAKE_C_FLAGS'] += [f'--gcc-install-dir={self.gcc_prefix}']
+            self.runtimes_cmake_args['CMAKE_CXX_FLAGS'] += [f'--gcc-install-dir={self.gcc_prefix}']
```

Comment:
I would still make sure this is called every time we used to call `_set_gcc_prefix`.
Right now we never reset `runtimes_cmake_args` across stages, but in case we ever start doing it, that would prevent shooting ourselves in the foot

Reply:
We **append** to the flags here so calling it again would be a mistake, wouldn't it? If we reset it I'd say it would be unexpected to suddenly add it. That should be done on purpose / with clear intent, shouldn't it?

As for previous usages:
- `_create_compiler_config_file` shouldn't set CMake flags, should it? I see that in the iterations of staged builds this is called to the flags are indirectly set but also only once: By the `self._set_gcc_prefix()` here in configure step after which `self._set_gcc_prefix()` is a no-op
- When running `sanity_check_step` this won't be used anymore

So IMO the least surprising place to set them is here in `configure_step` which should be exactly the same behavior as before unless we end up skipping this step